src/objective-c/examples/Sample/SampleTests/RemoteProtoTests.m

```diff
@@ -269,4 +269,37 @@ - (void)testCancelAfterBeginRPC {
   [self waitForExpectationsWithTimeout:1 handler:nil];
 }
 
+- (void)testCancelAfterFirstResponseRPC {
+  __weak XCTestExpectation *expectation = [self expectationWithDescription:@"CancelAfterFirstResponse"];
+  
+  // A buffered pipe to which we never write any value acts as a writer that just hangs.
```

Comment:
But you _are_ writing some value! :) Instead, you can e.g. add a comment before calling cancel, saying explicitly that we don't tell the requestBuffer that we're done.